pocket_coffea/lib/jets.py

```diff
@@ -216,10 +216,13 @@ def jet_selection(events, jet_type, params, leptons_collection=""):
         dR_jets_lep = jets.metric_table(events[leptons_collection])
         mask_lepton_cleaning = ak.prod(dR_jets_lep > cuts["dr_lepton"], axis=2) == 1
 
-    if jet_type == "Jet":
+    if 'run_period' not in params or params['run_period']!='Run3': # puid does not exist for Run3 samples
```

Comment:
Let's discuss a moment about this approach. In my opinion the run_period is a charactheristic of the dataset, not of the parameters.  We should follow the same pattern as the `year` for many corrections, adding the `run_period` metadata key to the dataset or automatically assigning it from the `year` field.

This info is accessing from `events.metadata` and we can preload it in the `load_metadata()` function of the base processor like the year  https://github.com/PocketCoffea/PocketCoffea/blob/main/pocket_coffea/workflows/base.py#L121

Reply:
I agree with this. Just did not find how to get the info before. The events.metadata works, so switching to this.

I also agree that it would be good to have a single `run_period` attribute of a dataset in metadata, determined from `year`. But  I did not implement this now.

Reply:
Ok fine, I think the new implementation is fine and we can then elaborate more on how to handle the run_period automatically. Probably a new default parameters set will do.